clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -136,27 +369,160 @@ public ConfigDef withClientSaslSupport() {
      * the appropriate type (int, string, etc)
      */
     public Map<String, Object> parse(Map<?, ?> props) {
+        return parse(props, true);
+    }
+
+    /**
+     * Parse and validate configs against this configuration definition. The input is a map of configs. It is expected
+     * that the keys of the map are strings, but the values can either be strings or they may already be of the
+     * appropriate type (int, string, etc). This will work equally well with either java.util.Properties instances or a
+     * programmatically constructed map.
+     *
+     * @param props The configs to parse and validate
+     * @param checkRequired Whether to throw a config exception in case that we don't provide a value for a required config
+     * @return Parsed and validated configs. The key will be the config name and the value will be the value parsed into
+     * the appropriate type (int, string, etc)
+     */
+    public Map<String, Object> parse(Map<?, ?> props, boolean checkRequired) {
```

Comment:
Having looked through the rest of the patch, I don't understand why this `checkRequired` was added. It looks like it is always set to `true` anyway? Originally I thought this was for running `parse` just to do validation such that it could fail, but it seems it isn't used that way?